tokio-util/src/future.rs

```diff
@@ -0,0 +1,136 @@
+//! An extension trait for Futures that provides a variety of convenient adapters.
+
+mod with_cancellation_token;
+use with_cancellation_token::{WithCancellationTokenFuture, WithCancellationTokenFutureOwned};
+
+use std::future::Future;
+
+use crate::sync::CancellationToken;
+
+/// A trait which contains a variety of convenient adapters and utilities for `Future`s.
+pub trait FutureExt: Future {
+    /// A wrapper around [`tokio::time::timeout`], with the advantage that it is easier to write
+    /// fluent call chains.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use tokio::{sync::oneshot, time::Duration};
+    /// use tokio_util::future::FutureExt;
+    ///
+    /// # async fn dox() {
+    /// let (_tx, rx) = oneshot::channel::<()>();
+    ///
+    /// let res = rx.timeout(Duration::from_millis(10)).await;
+    /// assert!(res.is_err());
+    /// # }
+    /// ```
+    #[cfg(feature = "time")]
```

Comment:
Please use `cfg_time!` to render the feature gate in the `docs.rs`.

<img width="1585" height="622" src="https://github.com/user-attachments/assets/fa25275e-acfe-4c9a-85a7-00402e200a6f" />

We could refer the below example

https://github.com/tokio-rs/tokio/blob/ce41896f8dcbc6249df3279600f45f7a65915cf6/tokio-util/src/lib.rs#L51-L53